src/game/uo_files/CUOMobtypes.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+* @file CUOMobtypes.cpp
+*
+*/
+
+#include "../../sphere/threads.h"
+#include "../../common/CException.h"
+#include "../../common/CExpression.h"
+#include "../../common/CLog.h"
+#include "../../common/CUOInstall.h"
+#include "CUOMobtypes.h"
+
+void CUOMobTypes::Load()
+{
+    ADDTOCALLSTACK("CUOMobTypes::Load");
+    g_Log.Event(LOGM_INIT, "Caching mobtypes.txt...\n");
+
+    CUOMobTypesType mobTypesRow = {0,0};
+
+    _mobTypesEntries.clear();
+
+    CSFileText csvMobTypes;
+
+    if (g_Install.OpenFile(csvMobTypes, "mobtypes.txt", (word)(OF_READ | OF_TEXT | OF_DEFAULTMODE)))
+    {
+        _mobTypesEntries.resize(4096);
+        for (int i = 0; i < 4096; i++)
+        {
+            mobTypesRow.m_type = 4;
+            mobTypesRow.m_flags = 0;
+            _mobTypesEntries[i] = mobTypesRow;
+        }
+
+        tchar* pszTemp = Str_GetTemp();
+        size_t count = 0;
+        while (!csvMobTypes.IsEOF())
+        {
+            csvMobTypes.ReadString(pszTemp, 200);
+            if (*pszTemp)
+            {
+                count++;
+
+                std::string tmpString = pszTemp;
+
+                std::vector<std::string> splitArray;
+
+                int len = (int)tmpString.length();
+                len = Str_TrimEndWhitespace(tmpString.data(), len);
+
+                if (len == 0 || tmpString[0] == '#') //Empty line or commented
+                    continue;
+
+                //Split the string
+                char* pch;
+                pch = strtok(tmpString.data(), " \t");
```

Comment:
You could check, if you like so, Str_Parse. It's a Sphere function which hasn't the drawbacks of strtok, mainly it's not thread safe.

Reply:
Like this?
```
//Split the string
std::vector<tchar*> splitArray;
splitArray.resize(3);
size_t iQty = Str_ParseCmds(tmpString.data(), splitArray.data(), 3, " \t#");
```

Reply:
Yes, or you could just use tchar* ppColumns[3] in place of that vector